src/gpu/intel/jit/ngen/ngen_elf.hpp

```diff
@@ -183,26 +214,144 @@ class ELFCodeGenerator : public BinaryCodeGenerator<hw>
+                        uint8_t tag = DWARF_TAG::SUBPROGRAM;
+                        uint8_t hasChildren = 0;
+
+                        struct {
+                                uint8_t attr_name;
+                                uint8_t attr_form;
+                        } attributes[5] = {
+                                { DWARF_AT::NAME, DWARF_FORM::STRP },
+                                { DWARF_AT::DECL_FILE, DWARF_FORM::DATA1 },
+                                { DWARF_AT::DECL_LINE, DWARF_FORM::DATA4 },
+                                { DWARF_AT::DECL_COLUMN, DWARF_FORM::DATA1 },
+                                { 0, 0 },
+                        };
+                } subProgram;
+                struct __attribute__((packed)) {
+                        uint8_t code = 0;
+                } end;
+        } debugAbbrev;
+
+        struct DebugLine {
+                struct __attribute__((packed)) DebugLineHeader {
+                        uint32_t unit_length;
+                        uint16_t version = 5;
+                        uint8_t address_size = sizeof(void*);
+                        uint8_t segment_selector_size = 0;
+                        uint32_t header_length;
+                        uint8_t minimum_instruction_length = 1;
+                        uint8_t maximum_operations_per_instruction = 1;
+                        uint8_t default_is_stmt = 1;
+                        int8_t line_base = 0;
+                        uint8_t line_range = 1;
+                        uint8_t opcode_base = 1;
+//                         uint8_t standard_opcode_lengths[] = {};
+
+                        uint8_t directory_entry_format_count = 1;
+                        struct {
+                                uint8_t type;
+                                uint8_t form;
+                        } directory_entry_format[1] = {
+                                { DWARF_LNCT::PATH, DWARF_FORM::LINE_STRP },
+                        };
+                        uint8_t directories_count = 1;
+                        uint32_t dirName = strlen(__FILE__) + 1;
+
+                        uint8_t file_name_entry_format_count = 2;
+                        struct {
+                                uint8_t type;
+                                uint8_t form;
+                        } file_name_entry_format[2] = {
+                                { DWARF_LNCT::PATH, DWARF_FORM::LINE_STRP },
+                                { DWARF_LNCT::DIRECTORY_INDEX, DWARF_FORM::DATA1 },
+                        };
+                        uint8_t file_names_count = 2;
+                        uint32_t fileName0 = 0;
+                        uint8_t dirIndex0 = 0;
+                        uint32_t fileName1 = 0;
+                        uint8_t dirIndex1 = 0;
+                } header;
+        } debugLine;
+
+        struct {
+                const char fileName[strlen(__FILE__) + 1] = __FILE__;
+                const char dirName[strlen("oneDNN:") + 1] = "oneDNN:";
```

Comment:
Indentation within this file should use 4 spaces.

Reply:
Fixed, thanks.